presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -504,13 +509,50 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                     newJoinFilter.isPresent() == node.getFilter().isPresent() &&
                             (!newJoinFilter.isPresent() || areExpressionsEquivalent(newJoinFilter.get(), node.getFilter().get()));
 
+            Map<String, Symbol> dynamicFilters = ImmutableMap.of();
+            List<Expression> predicates = ImmutableList.of();
+            if (node.getType() == INNER && isEnableDynamicFiltering(session)) {
+                // New equiJoinClauses could potentially not contain symbols used in current dynamic filters.
```

Comment:
Extract this entire code to a separate method. It can return class a result with dynamic filters and predicates.